gulpfile.js

```diff
@@ -69,6 +70,37 @@ var webpackConfig = {
   plugins: gulpPlugins
 };
 
+gulp.task('test-unit', function (done) {
+    new KarmaServer({
+        configFile: __dirname + '/karma.conf.js'
+    }, done).start();
+});
+
+gulp.task('test-coverage', function (done) {
+    new KarmaServer({
+        configFile: __dirname + '/karma.conf.js',
+        reporters: ['mocha', 'coverage', 'threshold'],
+        singleRun: true,
+        webpack: {
```

Comment:
Does this do a deep merge with `configFile` ? If not, we should most likely manually merge this ourselves.
This may explain why the custom extension resolves are not working -

```javascript
+  resolve: {
+    extensions: ['', '.js', '.jsx']
+  }
```